docs/manual/iomode.md

```diff
@@ -0,0 +1,94 @@
+---
+layout: default
+title: rpm.org - RPMIO Mode
+---
+# RPMIO mode
+
+RPM supports a variety of compression methods through external libraries for
+its internal I/O operations.  For ease of use, the [RPMIO
+API](https://ftp.osuosl.org/pub/rpm/api/4.17.0/group__rpmio.html) offers
+special `fopen(3)/fdopen(3)`-like functions that accept an extended mode string
+which allows for specifying a compression method alongside the usual `r` or `w`
+letters.  RPM will then perform (de)compression transparently for the caller.
+
+While primarily intended for internal use, a mode string can also be exposed
+through configuration options or macros where appropriate.  Currently, only
+payload write mode can be configured this way, see the
+[Payload](format.html#payload) section for details.
+
+## Format
+
+The format of an RPMIO mode string is as follows:
+
+```
+<mode>[flags].<type>
+```
+
+The `mode` parameter is mandatory and contains a sequence of letters as
+described in `fopen(3)`.  For payload configuration, only `w` is allowed.
+
+The `type` parameter is also mandatory and specifies the compression type
+(*compressor*) to be used.  Currently, the following compressors are supported:
+
+| Type		| Description					    |
+|---------------|---------------------------------------------------|
+|`gzdio`	| gzip ([flags](#gzip-flags))			    |
+|`bzdio`	| bzip2 ([flags](#bzip2-flags))			    |
+|`xzdio`	| xz ([flags](#xz-flags))		    |
+|`lzdio`	| lzma (legacy) ([flags](#lzma-flags))  |
+|`zstdio`	| zstd ([flags](#zstd-flags))			    |
+|`ufdio`	| uncompressed					    |
+
+The `flags` parameter is optional and contains compressor-specific options,
+most commonly a compression *level* or the *number of threads*.  If an option
+is omitted, the compressor's default value is used.  For the actual supported
+values, consult the given compressor's section below.  For some real-world
+examples, see [Example modes](#example-modes).
+
+#### GZIP flags
+
+These can include a compression level (from 0 to 9) and a strategy (e.g. `h`
+for Huffman-only compression).  See the description of `gzopen()` in
+`/usr/include/zlib.h` for more details.
+
+#### BZIP2 flags
+
+These can include a compression level (from 1 to 9) and `s` (enables "small
```

Comment:
Oh okay, 's' is another compressor-specific (small) flag. The real problem with these is that they leak the API of the underlying library through librpmio, which we do not want. So even if we may currently let them pass through unnoticed, I think we should not document these, but only the bits explicitly supported by librpmio.